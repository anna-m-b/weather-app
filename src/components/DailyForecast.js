import React from 'react';
import PropTypes from 'prop-types';
import '../styles/DailyForecast.css';
import '../styles/weather-icons.min.css';

const DailyForecast = ({ date, icon, temperature, description }) => {
  return (
    <div className="DailyForecast" data-testid="daily-forecast">
      <div className="DailyForecast_date">{date}</div>
      <figure
        className="DailyForecast_icon"
        role="figure"
        aria-label={`${description} icon`}
      >
        <i className={icon} />
      </figure>
      <div className="DailyForecast_temperature">{`${temperature}Â°c`}</div>
      <div className="DailyForecast_description">{description}</div>
      <div className="DailyForecast_more-details">More details</div>
    </div>
  );
};

DailyForecast.propTypes = {
  date: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
};

export default DailyForecast;
