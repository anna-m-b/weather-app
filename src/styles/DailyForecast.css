.DailyForecast {
  margin: 5px;
  /* border: 0.5px solid blue; */
  border-radius: 15px;
  background-color: #ffffff38;
  width: 6.5rem;
  height: 9rem;
  display: grid;
  align-items: center;
  padding: 5px;
  font-size: 1rem;
  box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
  -webkit-box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
  -moz-box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
}

.DailyForecast_icon, .DailyForecast_date {
  justify-self: center;
}

.DailyForecast_temperature, .DailyForecast_description, .DailyForecast_more-details {
  padding-left: 0.6rem;
}

.DailyForecast_icon {
  font-size: 2.3rem;
  margin: 0.2rem;
  color: #160064;
}


.DailyForecast_date {
  font-weight: 500;
}

.DailyForecast_more-details {
  font-size: 0.6rem;
}