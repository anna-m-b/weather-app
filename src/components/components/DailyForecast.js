import React from 'react';
import '../../styles/DailyForecast.css';
import '../../styles/weather-icons.min.css';

const DailyForecast = () => {
  return (
    <div className="DailyForecast">
      <p className="date">Sat 6th Feb</p>
      <i className="wi wi-day-rain" />
      <p>
        6Â°c
        <br />
        Rainy
        <br />
        <span className="more-details">More details</span>
      </p>
    </div>
  );
};

export default DailyForecast;
