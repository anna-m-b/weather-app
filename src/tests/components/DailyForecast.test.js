import React from 'react';
import { render, screen } from '@testing-library/react';
import DailyForecast from '../../components/DailyForecast';

describe('DailyForecast', () => {
  const validProps = {
    date: 'Mon 8th Feb',
    icon: 'wi wi-day-rain',
    description: 'Rainy',
    temperature: 6,
  };

  it('renders correctly', () => {
    const { asFragment } = render(<DailyForecast {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('it renders each element based on props', () => {
    beforeEach(() => {
      render(
        <DailyForecast
          date="Mon 8th Feb"
          icon="wi wi-day-rain"
          description="Rainy"
          temperature={6}
        />
      );
    });

    it('renders the date', () => {
      expect(screen.getByText('Mon 8th Feb')).toBeTruthy();
      expect(screen.getByText('Mon 8th Feb')).toHaveClass('DailyForecast_date');
    });

    it('renders a weather icon', () => {
      expect(screen.getByRole('figure', { name: /rainy/i })).toBeTruthy();
      expect(screen.getByRole('figure', { name: /rainy/i })).toHaveClass(
        'DailyForecast_icon'
      );
    });

    it('renders the temperature', () => {
      expect(screen.getByText('6°c')).toBeTruthy();
      expect(screen.getByText('6°c')).toHaveClass('DailyForecast_temperature');
    });

    it('renders a weather description as text', () => {
      expect(screen.getByText('Rainy')).toBeTruthy();
      expect(screen.getByText('Rainy')).toHaveClass(
        'DailyForecast_description'
      );
    });
  });
});
