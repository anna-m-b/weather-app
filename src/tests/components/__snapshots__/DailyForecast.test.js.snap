// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForecast renders correctly 1`] = `
<DocumentFragment>
  <div
    class="DailyForecast"
    data-testid="daily-forecast"
  >
    <div
      class="DailyForecast_date"
    >
      Mon 8th Feb
    </div>
    <figure
      aria-label="Rainy icon"
      class="DailyForecast_icon"
      role="figure"
    >
      <i
        class="wi wi-day-rain"
      />
    </figure>
    <div
      class="DailyForecast_temperature"
    >
      6Â°c
    </div>
    <div
      class="DailyForecast_description"
    >
      Rainy
    </div>
    <div
      class="DailyForecast_more-details"
    >
      More details
    </div>
  </div>
</DocumentFragment>
`;
